codes/quantum/oscillators/dual_rail.yml

```diff
@@ -0,0 +1,53 @@
+code_id: dual_rail
+
+physical: qubits
+logical: qubits
+
+name: 'Dual Rail'
+
+introduced: \cite{doi:10.1103/PhysRevA.52.3489}
+
+description: 'In a dual rail qubit encoding, single qubits are represented by Fock states
+in 2 modes. The logical \(\ket{0}\) state is represented by \(\ket{01}\),
+while the logical \(\ket{1}\) state is represented by \(\ket{10}\). The original
+paper that introduced this encoding seeks to implement Deutsch''s algorithm specifically,
+i.e., it does not provide a framework for universal quantum computation (although this can
+be realized using Kerr non-linearities along with beamsplitters), and it does not solve the
+general problem of error correction.
+In the original paper
+that introduced this encoding, error correction is performed by simply rejecting
+executions which result in error states (i.e, illegal results), and re-executing.'
+
+protection: 'This code cannot protect against general errors.
```

Comment:
Either explain the Caldeira-Leggett coupling and ref to errors mixing modes in the context Deutsch's problem, or omit one or both. Simply put, this code detects one loss error for each mode, which maps one of the logical codewords into the vacuum state \(|00\rangle\).